### PR TITLE
Submodule

### PR DESCRIPTION
**Pull request by PiotrBanuba**

### CODE REVIEW

File: libraries/utils/ogl_utils/src/program.cpp [hunk 1]
```diff
@@ -56,7 +60,10 @@ program::program(const char* name, const char* vertex_shader_code, const char* f
     if (!success)
     {
         GL_CALL(glGetShaderInfoLog(fragmentShader, 512, NULL, infoLog));
-        throw std::runtime_error(infoLog);
+        std::string s("fragment shader:\n");
+        s.append(infoLog);
+        throw std::runtime_error(s.c_str());
+        //throw std::exception();
```
Comment (nick-korotysh-bnb) [inline libraries/utils/ogl_utils/src/program.cpp:66]: remove commented code

File: libraries/utils/ogl_utils/src/program.cpp [hunk 2]
```diff
@@ -69,6 +76,7 @@ program::program(const char* name, const char* vertex_shader_code, const char* f
     if (!success) {
         glGetProgramInfoLog(shaderProgram, 512, NULL, infoLog);
         throw std::runtime_error(infoLog);
+        //throw std::exception(infoLog);
```
Comment (nick-korotysh-bnb) [inline libraries/utils/ogl_utils/src/program.cpp:79]: remove commented code

File: offscreen_effect_player/pixel_buffer.hpp [hunk 3]
```diff
@@ -26,10 +26,10 @@ namespace bnb
         virtual void get_texture(oep_texture_cb callback) override;
     private:
         oep_wptr m_oep_ptr;
-        uint8_t lock_count = 0;
+        uint8_t lock_count {0};
 
-        uint32_t m_width = 0;
-        uint32_t m_height = 0;
+        uint32_t m_width {0};
```
Comment (nick-korotysh-bnb) [inline offscreen_effect_player/pixel_buffer.hpp:31]: maybe that was unnecessary, but I'm ok with it, even don't like such style in general (especially for such simple cases).
don't revert it, leave as is

File: offscreen_effect_player/yuv_converter.cpp [hunk 4]
```diff
@@ -0,0 +1,359 @@
...
+    }
+    throw std::runtime_error(msg);
+}
+
+namespace bnb::converter
+{
+
+const int drawing_plane_vert_count = 4;
+
+#define N "\n"
```
Comment (nick-korotysh-bnb) [inline offscreen_effect_player/yuv_converter.cpp:46]: don't like such defines, I don't think that it is worth to save only one character

File: offscreen_effect_player/yuv_converter.hpp [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+        uint32_t m_in_texture_uniform {0};
+        int32_t m_draw_indent {0};
+    };
+
+
+
+    /* yuv_converter::get_width */
+    inline int yuv_converter::get_width() {
+        return m_width;
+    }
```
Comment (nick-korotysh-bnb) [inline offscreen_effect_player/yuv_converter.hpp:102]: you can write such thing in just one line, but this is just a not. I'm fine with this style

**Review by nick-korotysh-bnb — APPROVED** [review]
didn't review GL-related stuff because don't any expertise in this

**Review by dmitrystoyanovsky — APPROVED** [review]

File: offscreen_effect_player/yuv_converter.cpp [hunk 6]
```diff
@@ -0,0 +1,356 @@
...
+    float y_step {0.0f};
+    switch (m_rotation) {
+        case rotation::deg_0:
+            x_step = 1.0f / m_width;
+            break;
+        case rotation::deg_90:
+            y_step = (m_vertical_flip ? 1.0f : -1.0f) / m_width;
+            break;
+        case rotation::deg_180:
+            x_step = -1.0f / m_width;
```
Comment (dmitrystoyanovsky) [inline offscreen_effect_player/yuv_converter.cpp]: Check the spikes on the resulted image with that value of step for this orientation - I see spikes in my case and fixed that by removing - sign.

**Pull request merged**